core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -195,8 +203,108 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
+    try {
+      if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
+        throw new NotFoundException("Unknown namespace " + namespace);
+      }
+
+      for (FileStatus s : fs.listStatus(nsPath)) {
+        Path path = s.getPath();
+        if (!fs.isDirectory(path)) {
+          // Ignore the path which is not a directory.
+          continue;
+        }
+        Path metadataPath = new Path(path, "metadata");
+        if (fs.exists(metadataPath) && fs.isDirectory(metadataPath) &&
+            (fs.listStatus(metadataPath, TABLE_FILTER).length >= 1)) {
+        // Ignore the path of table
+          continue;
+        }
+
+        if (fs.exists(path) && fs.isDirectory(path)) {
+          // Only the path which contains metadata is the path for table, otherwise it could be
+          // still a namespace.
+          if (!namespace.isEmpty()) {
+            nsp = (namespace.toString() + "." + path.getName()).split("\\.");
+          } else {
+            nsp = new String[] {path.getName()};
+          }
+          namespaces.add(Namespace.of(nsp));
+        }
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to list namespace under " + namespace, ioe);
+    }
+
+    return namespaces;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    Preconditions.checkArgument(listTables(namespace).size() == 0,
+        "This Namespace have tables, cannot drop it");
+    Preconditions.checkArgument(listNamespaces(namespace).size() == 0,
+        "This Namespace have sub Namespace, cannot drop it");
+
+    Joiner slash = Joiner.on("/");
+    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    io().deleteFile(nsPath.toString());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    Joiner slash = Joiner.on("/");
+    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    Map<String, String> meta = new HashMap<>();
+    FileSystem fs = Util.getFs(nsPath, conf);
+    try {
+      if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
+        throw new NotFoundException("Unknown namespace " + nsPath);
+      }
+      FileStatus info = fs.getFileStatus(nsPath);
+      meta.put("owner", info.getOwner());
+      meta.put("group", info.getGroup());
+      meta.put("path", info.getPath().toString());
+      meta.put("modification_time", Long.toString(info.getModificationTime()));
+      meta.put("block_size", Long.toString(info.getBlockSize()));
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to list namespace info " + namespace, ioe);
+    }
+    return meta;
+  }
+
   @Override
   public void close() throws IOException {
   }
 
+  public FileIO io() {
```

Comment:
This should use the Hadoop `FileSystem`, not `FileIO.`

Reply:
thanks Ryan , now has fixed it.